src/telemetry-data/data-model.md

```diff
@@ -0,0 +1,203 @@
+# Thin-edge data model
+
+## Motivations
+
+* Move telemetry data around thin-edge components: the devices, the cloud, the sensors, the child devices, the analytics engines ...
+* Ease the ingestion of telemetry data coming from various sources, possibly implemented by tiers.
+* Ease the transfer of telemetry data to various cloud.
+* Ease the exchange of telemetry data between processing component running on a thin-edge device.
+
+## Current state of thin-edge JSON version 0.5
+
+Benefits:
+* Simple to learn, yet powerful
+* Requires no configuration
+* Atomic messages can be grouped into larger messages
+  * making the transfers more efficient,
+  * providing a better UX on the cloud console,
+  * and reducing the customer's bill (based on the message rate).
+* Writing a mapper between thin-edge JSON and another format is simple,
+  * as long as there is one-one correspondence between the messages.
+
+Limitations:
+* Many data sources *already* exist and produce data that can be mapped to thin-edge JSON,
+  * but are not *actually* thin-edge JSON,
+  * making mandatory yet another data mapper.
+* Thin-edge JSON is focused on a format, while what we need first is a data model.
+* The current format parser is too rigid,
+  * rejecting the whole message when only one part is not meaningful.
+* Many data sources are publishing data with one topic per kind of data
+  * while thin-edge expect all the data on a single topic.
+* Writing a mapper that groups messages as those produced by `collectd` is a more complex task than expected.
+  * One should leverage the effort done for `collectd` and use it 
+* All the data published on `tedge/measurements` are sent to the cloud.
+  * What if we wish to filter some data out?
+  * What if we wish to pre-process the data to only forward an aggregate?
+* The meaning and purpose of multi-measurements is not clear.
+* Possible name collisions with reserved key words (`time` and `type`)?
+* Non numeric values are rejected.
```

Comment:
This seems to be pretty big limitation to the format and as far as I know there are already some users asking why they can't send non numeric data, eg colour or encoded measurement stream.
Generic data model should not limit type of measurements sent.

Reply:
@makr11st 
But isn't numeric or non-numeric the differentiation between measurements and events? (At least in c8y world)

Reply:
This is great question.
In my world and where I worked before this is not how you differentiate between measurements and events.

A `measurement` is a metric collected at some interval, let say measure temperature every 1 second or count pulses of a meter every 10ms and output the total every 1 second. Basically a collected repeated observation of a state - it can be of any type, numeric value, string value, or binary stream if frequency is high. 

An `event` is something that happens due to some circumstances change, eg. new connection to the server or a device which collects gps coordinates and has geo fence configured gets close proximity to the fence it may create an event containing some measurements (eg coordinates) and additional metadata, let say information that we are less than 10m of the geofence.

Reply:
Agree with @makr11st. Measurements is about repeated observations of some state - more usually a number but why not some enumerated values. The same state can be observed at different pace and it make sense to infer non-observed values. Events happen independently of any observation and it makes no sense to infer an event that has been missed.

A confusion comes from the fact the a series of observations can be seen as a series of events (the value x has been observed at t for the metric m). For instance, [Cumulocity modelizes location updates as events](https://cumulocity.com/guides/device-sdk/mqtt/#401).

So I think one must let the application developer free to use numbers or opaque string values both for measurements and events.